CMakeLists.txt

```diff
@@ -57,23 +61,47 @@ message(STATUS "CUTLASS ${_CUTLASS_VERSION_MAJOR}.${_CUTLASS_VERSION_MINOR}.${_C
 
 project(CUTLASS VERSION ${_CUTLASS_VERSION_MAJOR}.${_CUTLASS_VERSION_MINOR}.${_CUTLASS_VERSION_PATCH} LANGUAGES CXX)
 
-################################################################################
+###################################################################################################
+#
+# Configure SYCL build options
+#
+###################################################################################################
+
+set(CUTLASS_ENABLE_SYCL OFF CACHE BOOL "Enable SYCL")
 
-include(${CMAKE_CURRENT_SOURCE_DIR}/CUDA.cmake)
+if (CUTLASS_ENABLE_SYCL)
+  set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
+  if("${DPCPP_SYCL_TARGET}" STREQUAL "nvptx64-nvidia-cuda")
+    set(SYCL_NVIDIA_TARGET ON)
+    add_compile_definitions(SYCL_NVIDIA_TARGET)
+  elseif("${DPCPP_SYCL_TARGET}" STREQUAL "intel_gpu_pvc")
+    set(SYCL_INTEL_TARGET ON)
+    add_compile_definitions(SYCL_INTEL_TARGET)
+  endif()
 
-if (CUDA_VERSION VERSION_LESS 11.3)
-  message(WARNING "CUTLASS ${CUTLASS_VERSION} requires CUDA 11.4 or higher, and strongly recommends CUDA 11.8 or higher.")
-elseif (CUDA_VERSION VERSION_LESS 11.4)
-  message(WARNING "CUTLASS ${CUTLASS_VERSION} support for CUDA ${CUDA_VERSION} is deprecated, please use CUDA 11.8 or higher.")
+  add_compile_definitions(CUTLASS_ENABLE_SYCL)
+  find_package(DPCPP)
+  include(${CMAKE_CURRENT_SOURCE_DIR}/SYCL.cmake)
+
+else()
+  include(${CMAKE_CURRENT_SOURCE_DIR}/CUDA.cmake)
+
+  if (CUDA_VERSION VERSION_LESS 11.3)
+    message(WARNING "CUTLASS ${CUTLASS_VERSION} requires CUDA 11.4 or higher, and strongly recommends CUDA 11.8 or higher.")
+  elseif (CUDA_VERSION VERSION_LESS 11.4)
+    message(WARNING "CUTLASS ${CUTLASS_VERSION} support for CUDA ${CUDA_VERSION} is deprecated, please use CUDA 11.8 or higher.")
+  endif()
+
+  if (CUDA_COMPILER MATCHES "[Cc]lang" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.0)
+    message(FATAL_ERROR "Clang 7.0+ required for GPU compilation")
+  endif()
 endif()
 
 if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.3)
   message(FATAL_ERROR "GCC version must be at least 7.3!")
```

Comment:
Is this for both SYCL and CUDA or is it only required for cuda as far as I know we dont use GCC